shopping_mall/backend/ai/agent/executor.py

```
@@ -0,0 +1,677 @@
+            return "상품 정보 조회 중 오류가 발생했습니다."
+
+    async def _tool_search_farm_info(self, query: str) -> str:
+        docs = self.rag.retrieve(query, "farm_intro", top_k=3, distance_threshold=0.50)
+        if not docs:
+            return (
+                "FarmOS는 검증된 농장의 신선 농산물을 산지 직송으로 연결하는 플랫폼입니다. "
+                "유기농·친환경 인증 상품을 중심으로 엄선된 농가와 협력하고 있습니다."
+            )
+        return "\n\n".join(docs)
+
+    def _tool_escalate_to_agent(self, reason: str, urgency: str = "normal") -> str:
```

Ruff (0.15.10)

[warning] Unused method argument: `reason`

(ARG002)